metomi/rose/apps/rose_bunch.py

```diff
@@ -554,3 +559,36 @@ def same_prev_config(self, current):
         if current:
             unchanged = False
         return unchanged
+
+
+def simplify_path(path):
+    """Removes duplication in paths whilst maintaining integrity.
+
+    If duplicate items are present in a path this keeps the first item and
+    removes any subsequent duplicates.
+
+    Examples:
+        >>> simplify_path('')
+        ''
+        >>> simplify_path('a')
+        'a'
+        >>> simplify_path('a:a:a')
+        'a'
+        >>> simplify_path('a:b:a')
+        'a:b'
+        >>> simplify_path('a:b:b:a')
+        'a:b'
+        >>> simplify_path('a:b:a:b:c:d:a:b:c:d:e')
+        'a:b:c:d:e'
+
+    """
+    path = path.split(':')
+    counter = Counter(path)
+    for item, count in counter.items():
+        ptr = len(path) - 1
+        while count > 1:
+            if path[ptr] == item:
+                path.pop(ptr)
+                count -= 1
+            ptr -= 1
```

Comment:
Simpler might be
```suggestion
    path = list(dict.fromkeys(path))
```

> As of Python 3.7 (and CPython 3.6), standard dict is [guaranteed to preserve order](https://stackoverflow.com/questions/39980323/are-dictionaries-ordered-in-python-3-6)

https://stackoverflow.com/a/53657523/3217306

Reply:
Much nicer, can be a one-liner.